libraries/chain/block_header_state.cpp

```diff
@@ -78,7 +120,8 @@ const vector<digest_type>& block_header_state::get_new_protocol_feature_activati
 void finish_next(const block_header_state& prev,
                  block_header_state& next_header_state,
                  vector<digest_type> new_protocol_feature_activations,
-                 instant_finality_extension if_ext) {
+                 instant_finality_extension if_ext,
+                 std::optional<finalizer_policy> new_finalizer_policy) {
```

Comment:
I would prefer not pass this additional parameter (there will be two when we do `proposer_policies` as well).
I understand that it would require an extra `apply_diff` for the producer, but I don't think it would make any significant difference, and the code would be cleaner and more streamlined. The `calculate_finalizer_policy` could be moved to `finish_next`.